src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,113 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] arrayList;
+    private int size;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        reviewSize();
+        arrayList[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < size && index >= 0) {
+            reviewSize();
+            addingShift(index);
+            arrayList[index] = value;
+        } else {
+            throw new ArrayIndexOutOfBoundsException();
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        return (T) arrayList[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < size && index >= 0) {
+            arrayList[index] = value;
+        } else {
+            throw new ArrayIndexOutOfBoundsException();
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T nedded = get(index);
+        removingShift(index);
+        return nedded;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        int index = -1;
+        for (int i = 0; i < arrayList.length; i++) {
+            if (t == arrayList[i] || (t != null && t.equals(arrayList[i]))) {
+                index = i;
```

Comment:
```suggestion
                return.remove(i);
```
that will simplify your code